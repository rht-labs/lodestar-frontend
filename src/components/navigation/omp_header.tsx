import React from 'react';

import {
  PageHeader,
  Brand,
  Toolbar,
  ToolbarItem,
  ToolbarGroup,
  Nav,
  NavItem,
  NavList,
} from '@patternfly/react-core';

import { Link, useLocation } from 'react-router-dom';

import {
  CogsIcon,
  ListUlIcon,
  HomeIcon,
  HandPaperIcon,
} from '@patternfly/react-icons';

import { UserDropdown } from '../user_dropdown';

function _OMPHeader() {
  const iconPad: React.CSSProperties = {
    paddingBottom: 0,
    paddingTop: 0,
  };

  const { pathname } = useLocation();

  return (
    <PageHeader
      showNavToggle
      logo={
        <div>
          <Toolbar>
            <Brand
              alt="Open Innovation Labs"
              src={`${process.env.PUBLIC_URL}/oil_logo.png`}
            ></Brand>
            <div style={{ width: 50 }} />
          </Toolbar>
        </div>
      }
      headerTools={
        <Toolbar>
          <ToolbarGroup>
            <Nav theme="dark" variant="horizontal">
              <NavList>
                <NavItem
                  id="homeLink"
                  itemId={0}
                  isActive={pathname === '/dashboard'}
                >
                  <Link to="/app" style={iconPad}>
                    <HomeIcon size="md" title="Home" />
                  </Link>
                </NavItem>
                <NavItem
                  id="engagementPaneLink"
                  itemId={1}
                  isActive={pathname === '/engagements'}
                >
                  <Link style={iconPad} to="/app/engagements">
                    <ListUlIcon size="md" title="Engagements" />
                  </Link>
                </NavItem>
                <NavItem
                  id="adminLink"
                  itemId={3}
                  isActive={pathname === '/admin'}
                >
                  <Link style={iconPad} to="/app/admin">
                    <CogsIcon size="md" title="Administration" />
                  </Link>
                </NavItem>
              </NavList>
            </Nav>
          </ToolbarGroup>
          <ToolbarGroup>
            <ToolbarItem>
              <a
                href="https://docs.google.com/forms/d/e/1FAIpQLSfcKY5eKwDYSxIF9oYeDDVyYCqwcq_AD0eqhY4uLtpcCgfWwA/viewform"
                target="_blank"
                rel="noopener noreferrer"
              >
                <HandPaperIcon size="md" title="Feature Request" />
              </a>
            </ToolbarItem>
          </ToolbarGroup>
          <ToolbarGroup>
            <ToolbarItem>
              <UserDropdown />
            </ToolbarItem>
          </ToolbarGroup>
        </Toolbar>
      }
    ></PageHeader>
  );
}

export const OMPHeader = React.memo(_OMPHeader);
