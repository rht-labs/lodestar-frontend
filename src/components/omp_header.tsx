import React from 'react';

import {
  PageHeader,
  Brand,
  Toolbar,
  ToolbarItem,
  ToolbarGroup,
  Nav,
  NavItem,
  NavList,
  NavVariants,
  Avatar,
} from '@patternfly/react-core';  

import {
  Link,
  useLocation
} from 'react-router-dom';
        
import {
  CogsIcon,
  ListUlIcon,
  HomeIcon,
  HandPaperIcon,
} from '@patternfly/react-icons';

import avatarImg from '../assets/images/avatar.svg';
import { UserDropdown } from './user_dropdown';

function _OMPHeader() {
  const iconPad: React.CSSProperties = {
    paddingBottom: 0,
    paddingTop: 0,
  };

  const { pathname } = useLocation();

  return (
    <PageHeader
      logo={
        <div>
          <Toolbar>
            <Brand
              alt="Open Innovation Labs"
              src={`${process.env.PUBLIC_URL}/oil_logo.png`}
            ></Brand>
            <div style={{ width: 50 }} />
          </Toolbar>
        </div>
      }
      toolbar={
        <Toolbar>
          <ToolbarGroup>
            <Nav theme="dark">
              <NavList variant={NavVariants.horizontal}>
                <NavItem id="homeLink" itemId={0} isActive={pathname === '/'}>
                  <Link to="/" style={iconPad}>
                    <HomeIcon size="md" title="Home" />
                  </Link>
                </NavItem>
                <NavItem id="engagementPaneLink" itemId={1} isActive={pathname === '/engagements'}>
                  <Link style={iconPad} to="/engagements">
                    <ListUlIcon size="md" title="Engagements" />
                  </Link>
                </NavItem>
                <NavItem id="adminLink" itemId={3} isActive={pathname === '/admin'}>
                  <Link style={iconPad} to="/admin">
                    <CogsIcon size="md" title="Administration" />
                  </Link>
                </NavItem>
              </NavList>
            </Nav>
          </ToolbarGroup>
          <ToolbarGroup>
            <ToolbarItem>
              <a
                href="https://docs.google.com/forms/d/e/1FAIpQLSfcKY5eKwDYSxIF9oYeDDVyYCqwcq_AD0eqhY4uLtpcCgfWwA/viewform"
                target="_blank"
                rel="noopener noreferrer"
              >
                <HandPaperIcon size="md" title="Feature Request" />
              </a>
            </ToolbarItem> 
          </ToolbarGroup>
          <ToolbarGroup>
            <ToolbarItem>
              <UserDropdown />
            </ToolbarItem>
          </ToolbarGroup>
        </Toolbar>
      }
      avatar={<Avatar src={avatarImg} alt={'User Avatar'} />}
    ></PageHeader>
  );
}

export const OMPHeader = React.memo(_OMPHeader);