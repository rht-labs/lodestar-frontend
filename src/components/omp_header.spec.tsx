import React from 'react';
import { OMPHeader } from './omp_header';

describe('OMP Header UI', () => {
  test('should match snapshot', () => {
    expect(<OMPHeader />).toMatchSnapshot();
  });
});
