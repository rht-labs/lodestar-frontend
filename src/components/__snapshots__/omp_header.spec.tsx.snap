// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OMP Header UI should match snapshot 1`] = `<UNDEFINED />`;
